astrokat/observe_main.py

```diff
@@ -365,134 +403,158 @@ def run_observation(opts, kat):
-                )
-        # List sources and their associated functions from observation tags
-        not_cals_filter_list = []
-        for cal_type in cal_tags:
-            not_cals_filter_list.append("~{}".format(cal_type))
-            cal_array = [cal.name for cal in catalogue.filter(cal_type)]
-            if len(cal_array) < 1:
-                continue  # do not display empty tags
-            user_logger.info(
-                "{} calibrators are {}".format(str.upper(cal_type[:-3]), cal_array)
-            )
-        user_logger.info(
-            "Observation targets are [{}]".format(
-                ", ".join(
-                    [
-                        repr(target.name)
-                        for target in catalogue.filter(not_cals_filter_list)
-                    ]
-                )
-            )
-        )
-
-        # TODO: the description requirement in sessions should be re-evaluated
-        # since the schedule block has the description
-        # Description argument in instruction_set should be retired, but is
-        # needed by sessions
-        # Assign proposal_description if available, else create a dummy
-        if "description" not in vars(opts):
-            session_opts = vars(opts)
-            description = "Observation run"
-            if "proposal_description" in vars(opts):
-                description = opts.proposal_description
-            session_opts["description"] = description
-
+    # TODO: the description requirement in sessions should be re-evaluated
+    # since the schedule block has the description
+    # Description argument in instruction_set should be retired, but is
+    # needed by sessions
+    # Assign proposal_description if available, else create a dummy
+    if "description" not in vars(opts):
+        session_opts = vars(opts)
+        description = "Observation run"
+        if "proposal_description" in vars(opts):
+            description = opts.proposal_description
+        session_opts["description"] = description
+
+    nr_obs_loops = len(obs_plan_params["observation_loop"])
+    with start_session(kat.array, **vars(opts)) as session:
+        session.standard_setup(**vars(opts))
+
+        # Each observation loop contains a number of observation cycles over LST ranges
+        # For a single observation loop, only a start LST and duration is required
         # Target observation loop
-        with start_session(kat.array, **vars(opts)) as session:
-            session.standard_setup(**vars(opts))
+        observation_timer = time.time()
+        for obs_cntr, observation_cycle in enumerate(obs_plan_params["observation_loop"]):
+            if nr_obs_loops > 1:
+                user_logger.info("Observation loop {} of {}."
+                                 .format(obs_cntr + 1, nr_obs_loops))
+                user_logger.info("Loop LST range {}."
+                                 .format(observation_cycle["LST"]))
```

Comment:
Instead of using `.format()` str manipulation logic within the logger, you could use the recommend str formatting for Python logs.

Suggestion:

```python
user_logger.info('Post with id %s not found', post_id)
```

The above is more efficient and less expensive.

Below I added a screenshot, highlighting the cost per log string formatting. Notice the native `%s` string formatting is 24ns faster than concatenating/adding 2/more string together and 124ns faster than using `.format()` method. :wink:

![https://user-images.githubusercontent.com/7910856/128380755-362d05b6-cebb-4e6f-893f-4c50f8469a4f.png](https://user-images.githubusercontent.com/7910856/128380755-362d05b6-cebb-4e6f-893f-4c50f8469a4f.png)

Some recommended reading:

- [Stackoverflow: Python - Logger string formatting](https://stackoverflow.com/a/13375482)
- [Python Logging Best Practices](https://coralogix.com/blog/python-logging-best-practices-tips/)